### PR TITLE
feat(core_crypto): add parallelized pfpks with lwe ciphertext list

### PR DESCRIPTION
**Pull request by soonum**
closes: _please link all relevant issues_

### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: tfhe/src/core_crypto/algorithms/test/lwe_private_functional_packing_keyswitch.rs [hunk 1]
```diff
@@ -79,23 +80,17 @@ fn test_parallel_pfpks_equivalence<Scalar: UnsignedTorus + Send + Sync>(
...
-            println!("serial:{elapsed_serial:?}");
 
-            let start_parallel = std::time::Instant::now();
             par_private_functional_keyswitch_lwe_ciphertext_into_glwe_ciphertext(
                 &lwe_pfpksk,
                 &mut output_glwe_parallel,
                 &random_lwe,
             );
-            let elapsed_parallel = start_parallel.elapsed();
-            println!("parallel:{elapsed_parallel:?}");
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/algorithms/test/lwe_private_functional_packing_keyswitch.rs:94]: 😨 was that here before ?

File: tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs [hunk 2]
```diff
@@ -286,3 +286,79 @@ pub fn private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphert
...
+    output: &mut GlweCiphertext<OutputCont>,
+    input: &LweCiphertextList<InputCont>,
+    thread_count: ThreadCount,
+) where
+    Scalar: UnsignedTorus,
+    KeyCont: Container<Element = Scalar>,
+    InputCont: Container<Element = Scalar>,
+    OutputCont: ContainerMut<Element = Scalar>,
+{
+    assert!(input.lwe_ciphertext_count().0 <= output.polynomial_size().0);
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs:340]: missing an assert on the thread count ? maybe it's missing in the other implem as well

File: tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs [hunk 3]
```diff
@@ -286,3 +286,79 @@ pub fn private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphert
...
+            thread_count,
+        );
+        buffer
+            .as_mut_polynomial_list()
+            .iter_mut()
+            .for_each(|mut poly| {
+                polynomial_wrapping_monic_monomial_mul_assign(&mut poly, MonomialDegree(degree));
+            });
+        slice_wrapping_add_assign(output.as_mut(), buffer.as_ref());
+    }
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs:363]: that's not what we do for the packing keyswitch to better utilize threads, see https://github.com/zama-ai/tfhe-rs/blob/a5579532bed2f0ab52d5c73db926a354b96877ff/tfhe/src/core_crypto/algorithms/lwe_packing_keyswitch.rs#L643

notably line 721 and following where we use parallel iterators for much better performance

**Review by IceTDrinker — COMMENTED** [review]
I think we might have misunderstood each other, unless I'm missing something

File: tfhe/src/core_crypto/algorithms/test/lwe_private_functional_packing_keyswitch.rs (hunk 1 above)
Comment (soonum) [reply tfhe/src/core_crypto/algorithms/test/lwe_private_functional_packing_keyswitch.rs:94]: Yep

File: tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs (hunk 2 above)
Comment (soonum) [reply tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs:340]: This function calls `par_private_functional_keyswitch_lwe_ciphertext_into_glwe_ciphertext_with_thread_count()` which contains the required `assert!`

Comment (soonum) [reply tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs:340]: Fixed with implementation of suggestion below.

**Review by IceTDrinker — APPROVED** [review]
Thanks a lot !

**Pull request merged**